hw06-atm/src/main/java/ru/otus/atm/ATM.java

```diff
@@ -0,0 +1,87 @@
+package ru.otus.atm;
+
+import com.google.common.collect.Lists;
+import ru.otus.banknotes.Banknotable;
+import ru.otus.banknotes.NominalValue;
+import ru.otus.banknotes.factory.BanknoteFactory;
+import ru.otus.exceptions.IncorrectWithdrawalValue;
+import ru.otus.exceptions.NoSuchNominalValueExistsException;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class ATM implements CashMachinable {
+    Map<NominalValue, Integer> existsCountOfNominalMap;
```

Comment:
Поля всегда приватные.
И название поля я бы поправил.
```suggestion
    private Map<NominalValue, Integer> nominalsCount;
```

`NominalValue`, кстати, можно назвать просто `Nominal`.